include/nbl/builtin/glsl/utils/swapchain_transform.glsl

```diff
@@ -0,0 +1,146 @@
+#ifndef _NBL_BUILTIN_GLSL_SWAPCHAIN_TRANSFORM_INCLUDED_
+#define _NBL_BUILTIN_GLSL_SWAPCHAIN_TRANSFORM_INCLUDED_
+
+#include "nbl/builtin/glsl/utils/swapchain_transform_e.h"
+
+// TODO test transforms outside of horizontal mirror rotate 180 (flip y)
+
+ivec2 nbl_glsl_swapchain_transform_forward(in uint swapchainTransform, in ivec2 coord, in ivec2 lastTexel) {
+    switch (swapchainTransform) 
+    {
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_IDENTITY:
+        return coord;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_90:
+        return ivec2(coord.y, lastTexel.x - coord.x);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_180:
+        return ivec2(lastTexel) - coord;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_270:
+        return ivec2(lastTexel.y - coord.y, coord.x);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR:
+        return ivec2(lastTexel.x - coord.x, coord.y);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_90:
+        return coord.yx;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_180:
+        return ivec2(coord.x, lastTexel.y - coord.y);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_270:
+        return lastTexel - coord.yx;
+    default:
+        return ivec2(0);
+    }
+}
+
+ivec2 nbl_glsl_swapchain_transform_backward(in uint swapchainTransform, in ivec2 coord, in ivec2 lastTexel) {
```

Comment:
fold this function into preTransform and set ivec2 lastTexel = screenSize - ivec2(1); as a local var